pyproject.toml

```diff
@@ -8,10 +8,10 @@ packages = [{include = "honeycomb", from = "src" }]
 
 [tool.poetry.dependencies]
 python = "^3.7, >= 3.7.2"
-opentelemetry-api = "^1.14.0"
-opentelemetry-sdk = "^1.14.0"
-opentelemetry-exporter-otlp = "^1.14.0"
-opentelemetry-instrumentation = "^0.36b0"
+opentelemetry-api = "^1.16.0"
```

Comment:
All these dependency updates helped us get over version conflicts that kept cropping up in this work... but I wonder if we should merge those in separately as a standalone PR, then rebase this to pull those changes in?